src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,245 @@
+            System.out.println(DIVIDER);
+            try {
+                switch (commandArgs.get("command").toLowerCase()) {
+                case "list":
+                    list();
+                    break;
+                case "done":
+                    setTaskDone(Integer.parseInt(commandArgs.get("param")), true);
+                    break;
+                case "delete":
+                    deleteTask(Integer.parseInt(commandArgs.get("param")));
+                    break;
+                case "todo":
+                    addTodo(commandArgs.get("param"), true);
+                    break;
+                case "deadline":
+                    addDeadline(commandArgs.get("param"), commandArgs.get("by"), true);
+                    break;
+                case "event":
+                    addEvent(commandArgs.get("param"), commandArgs.get("at"), true);
+                    break;
+                case "bye":
+                    exit = true;
+                    bye();
+                    break;
+                default:
+                    printUnknownCommand();
+                    break;
+                }
+            } catch (MissingArgumentException e) {
+                printWithIndent(e.getMessage(), 5);
+            } catch (NumberFormatException e) {
+                printWithIndent("☹ OOPS!!! Index must be a number.", 5);
+            } catch (IndexOutOfBoundsException e) {
+                String indexErrorMsg = String.format("☹ OOPS!!! Item of index %s does not exist in tasks.",
+                        commandArgs.get("param"));
+                printWithIndent(indexErrorMsg, 5);
+            } finally {
+                try {
+                    saveTasks();
+                } catch (IOException e) {
+
+                }
+            }
+
+            System.out.println(DIVIDER);
+
+            if (exit) {
+                break;
+            } else {
+                System.out.println("");
+            }
+        }
+
+        scan.close();
+    }
+
+    public static HashMap<String, String> parseCommand(String command) {
+        String[] parts = command.split(" /");
+        HashMap<String, String> map = new HashMap<String, String>();
+        String[] commandParts = parts[0].split(" ", 2);
+        map.put("command", commandParts[0]);
+        map.put("param", commandParts.length > 1 ? commandParts[1] : "");
+        for (int i = 1; i < parts.length; i++) {
+            String[] argParts = parts[i].split(" ", 2);
+            if (argParts.length == 2) {
+                map.put(argParts[0], argParts[1]);
+            }
+        }
+        return map;
+    }
+
+    public static void printWithIndent(String text, int count) {
+        System.out.println(" ".repeat(count) + text);
+    }
+
+    public static void greet() {
+        System.out.println(DIVIDER);
+        printWithIndent("Hello! I'm Duke", 5);
+        printWithIndent("What can I do for you?", 5);
+        System.out.println(DIVIDER + "\n");
```

Comment:
Perhaps all the strings being printed in here could be stored in a constant variable